btc-transaction-helper.js

```diff
@@ -47,31 +48,79 @@ class BtcTransactionHelper{
         }
     }
 
-    async transferBtc(senderAddressInformation, receiverAddress, amountInBtc, data) {
+    /**
+     * 
+     * @param {string} address 
+     * @param {number} amountInBtc
+     * @returns 
+     */
+    async selectSpendableUTXOsFromAddress(address, amountInBtc) {
+
+        const utxos = await this.getUtxos(address);
+
+        const selected = [];
+        let accumulated = 0;
+
+        for(let i = 0; i < utxos.length && accumulated < amountInBtc; i++) {
+            selected.push(utxos[i]);
+            accumulated += utxos[i].amount;
+        }
+
+        // Necessary to avoid floating point operation issues
+        const change = satoshisToBtc(btcToSatoshis(accumulated) - btcToSatoshis(amountInBtc));
+
+        return {
+            utxos: selected,
+            change: change,
+        };
+
+    };
+
+    async getUtxos(address) {
+        return await this.nodeClient.getUtxos(address);
+    }
+
+    /**
+     * Transfers the btc amounts to the recipients specified in the `outputs`.
+     * Gets the spendable utxos of the `senderAddressInformation` up to the sum of the `amountInBtc`s specified in the `outputs`.
+     * @param {{address: string, privateKey: string, info?: any }} senderAddressInformation 
+     * @param {Array<{recipientAddress: string, amountInBtc: number}>} outputs 
+     * @param {Array<any>} data 
+     * @returns {string} btcTxHash
+     */
+    async transferBtc(senderAddressInformation, outputs, data) {
         try {
-            let fundAmount = Number(amountInBtc + this.btcConfig.txFee).toFixed(8);
-            const fundTxId = await this.nodeClient.fundAddress(
-                senderAddressInformation.address,
-                fundAmount
-            );
-            let fundingTx = await this.getTransaction(fundTxId);
-
-            let outputIndex = -1;
-            for (let i = 0; i < fundingTx.outs.length; i++) {
-                let outputAddress = this.getOutputAddress(fundingTx.outs[i].script);
-                if (outputAddress == senderAddressInformation.address) {
-                    outputIndex = i;
-                }
-            }
-            if (outputIndex == -1) {
-                throw new Error('funding transaction does not contain the funded adress');
+
+            const totalAmountInBtc = outputs.reduce((sum, output) => {
+                return sum + output.amountInBtc;
+            }, 0);
+
+            const fromAddress = senderAddressInformation.address;
```

Comment:
This Denotes That The `senderAddressInformation` Already Has Funds In It.. Right?

Reply:
Yes.